exec/csnex/lib/io.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.IO;
+
+namespace csnex.rtl
+{
+    public class FileObject: Object
+    {
+        public FileObject(Stream fs)
+        {
+            file = fs;
+        }
+
+        public override string toString()
+        {
+            return string.Format("<FILE:{0}>", file.ToString());
+        }
+
+        public Stream file;
+    };
+
+    public class io
+    {
+        private readonly Executor Exec;
+        public Cell stdin { get; private set; }
+        public Cell stdout { get; private set; }
+        public Cell stderr { get; private set; }
+
+        public io(Executor exe)
+        {
+            Exec = exe;
+            stdin = new Cell(new FileObject(Console.OpenStandardInput()));
+            stdout = new Cell(new FileObject(Console.OpenStandardOutput()));
+            stderr = new Cell(new FileObject(Console.OpenStandardError()));
+        }
+
+        private Stream check_file(object pf)
+        {
+            Stream f = ((FileObject)pf).file;
+            if (f == null) {
+                Exec.Raise("IoException.InvalidFile", "");
+            }
+            return f;
+        }
+
+
+        public void close()
+        {
+            object ppf = Exec.stack.Pop().Object;
+
+            Stream f = check_file(ppf);
+            if (f == null) {
+                return;
+            }
+            f.Close();
+            f = null;
+        }
+
+        public void flush()
+        {
+            object ppf = Exec.stack.Pop().Object;
+
+            Stream f = check_file(ppf);
+            if (f == null) {
+                return;
+            }
+            f.Flush();
+        }
+
+        public void open()
+        {
+            Number mode = Exec.stack.Pop().Number;
+            string name = Exec.stack.Pop().String;
+
+            FileAccess fa;
+            FileMode fm;
+            switch ((Mode)Number.number_to_uint32(mode)) {
+                case Mode.read:
+                    fa = FileAccess.Read;
+                    fm = FileMode.Open;
+                    break;
+                case Mode.write:
+                    fa = FileAccess.Write;
+                    fm = FileMode.Append;
+                    break;
+            default:
```

Comment:
Indenting here should match the cases.

Reply:
Opps.  Seems no matter how you configure your IDEs formatting settings, it still tries really hard to format the code the way _it_ wants to format it.  :/  Fixed it.